Test suite for the gspread script

LogicAL007 opened the pull request:
-Tests for the connection with the Weather API.
-Tests to check if the expected column names are present.
 -Tests to check the data types of columns
-Tests to check if data in certain columns falls within valid ranges. 
-Tests related to successful worksheet creation.

LogicAL007 committed:
Test suite for the gspread script

-Tests for the connection with the Weather API.
-Tests to check if the expected column names are present.
-Tests to check the data types of columns
-Tests to check if data in certain columns falls within valid ranges.
-Tests related to successful worksheet creation.

elseagle commented on __pycache__/my_gspread.cpython-310.pyc:
Kindly delete/add binary files to .gitignore

elseagle commented on README.md:
```diff
@@ -1,77 +1,31 @@
 
 ## Prerequisites
 
-Before using this script, ensure you have the following prerequisites in place:
```
Why did you delete your former readme description?

LogicAL007 replied:
oh i didnt delete it, it has already been merged with the main branch after you approved it, this is probably because i edited a copy of it for this new readme sir

elseagle commented on my_gspread.py:
When you run your test, does it not trigger this whole script?

Try to use 
```
if __main__ == "__name__"

LogicAL007 replied:
okay sir

elseagle commented on my_gspread.py:
```diff
@@ -1,23 +1,59 @@
+# access keys using dotenv
 load_dotenv()
-API_key =  os.getenv('Project_key')
+gspread_key =  os.getenv('Project_key')
```
Use upper case for your environment variable names e.g PROJECT_KEY

LogicAL007 replied:
I will change that now sir

elseagle commented on my_gspread.py:
the name of this sheet is not good enough, my_spread is somewhat vague. If this is your main script or an util or helper? If it's your main or entry point, you can simply name this app or main.

elseagle commented on test_my_gspread.py:
```diff
@@ -0,0 +1,91 @@
+from my_gspread import pass_json_into_df
+
+load_dotenv()
+gspread_key = os.getenv('Project_key')
```
[Don't Repeat Yourself (DRY)](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

This is something you did in the `my_gspread.py` already. Why not create a setup() or connect_sheet() that you can always call?

elseagle commented on test_my_gspread.py:
```diff
@@ -0,0 +1,91 @@
+
+load_dotenv()
+gspread_key = os.getenv('Project_key')
+gc = gspread.service_account(gspread_key)
```
I think I'd prefer, you don't do this. When writing tests, a good practice is to avoid anything that will call APIs or connect to an external network outside your environment. So the way to go is to use mocks, If you try to run a test that connects over an API you're not unit-testing your code because you're now testing the code of the API inside your own code, the goal of this project is unit testing not integration testing

elseagle commented on test_my_gspread.py:
```diff
@@ -0,0 +1,91 @@
+    This test for the successful connection with the weather API, by checking the response state code
+     '''
+    payload = {'Key': config['Key'], 'q': 'berlin', 'aqi': 'no'}
+    r = requests.get("http://api.weatherapi.com/v1/current.json", params=payload)
```
Don't do this for unit tests, you're supposed to test units of your code

elseagle reviewed (changes requested):
Solid work, we're getting closer to closing this! Kindly check out the new comments

LogicAL007 committed:
This commit fixes the issues rasied, this can be found the description

-delete/add binary files to .gitignore
- ensure the test trigger this whole script
-usE upper case for your environment variable names e.g PROJECT_KEY
-Name this app or main
-Do not repeat yourself
-Do not connect your API for unit tests

LogicAL007 committed:
Data cleaning with python

-Data cleaning of the missing data
-Data cleaning of the inconsistent data

LogicAL007 committed:
delete the first commit to create a new pull request

elseagle commented:
You have merge conflicts to resolve, basically because you might have altered some files you worked on while this PR was open in another commit or branch and merged with the master/main branch. You need to update the two files affected to the version you need them to be in. After you've resolved it kindly merge.

LogicAL007 committed:
fix the merge issue
